person.rb

```diff
@@ -21,3 +36,38 @@ def can_use_services?
     of_age? || @parent_permission
   end
 end
+
+# Prepare base Decorator Make sure that it inherits from Nameable.
+class DecoratorBase < Nameable
+  def initialize(nameable_obj)
+    super()
+    @nameable = nameable_obj
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# Create a class that inherits from the base Decorator class.
+
+class CapitalizeDecorator < DecoratorBase
+  def correct_name
+    super.capitalize
+  end
+end
+
+# For the TrimmerDecorator: Create a class that inherits from the base Decorator class.
+
+class TrimmerDecorator < DecoratorBase
+  def correct_name
+    super[0, 10]
+  end
+end
```

Comment:
- [ ] Good job decorating your project, 👏. You created the `Nameable` class , the `CapitalizeDecorator` , the ` TrimmerDecorator `  and the  `DecoratorBase` directly in the Person file. Working this way make you violate the single responsibility principle.  It could be better to create a file for each class for the good practices reason 👍